src/components/carousel/index.js

```diff
@@ -145,11 +145,31 @@ export default class Carousel extends BaseComponent {
     return index;
   }
 
+  getSnapToOffsets = () => {
+    const {itemSpacings} = this.props;
+    const {containerWidth, pageWidth} = this.state;
+
+    if (containerWidth) {
+      const spacings = pageWidth === containerWidth ? 0 : itemSpacings;
+      const initialBreak = pageWidth - (containerWidth - pageWidth - spacings) / 2;
+      const snapToOffsets = _.times(presenter.getChildrenLength(this.props), index => initialBreak + index * pageWidth);
+      return snapToOffsets;
+    }
+  }
+
   shouldUsePageWidth() {
     const {loop, pageWidth} = this.props;
     return !loop && pageWidth;
   }
 
+  onContainerLayout = ({nativeEvent: {layout: {width: containerWidth}}}) => {
+    const update = {containerWidth};
```

Comment:
I would declare 'containerWidth' in the state object in the c'tor.

Reply:
This is how I started it, and them removed it on purpose. 
I want it to be undefined at the beginning so it can block the initial render (otherwise there's a flicker)

Reply:
Yes. Of course, as undefined. I only like to declare all state variables at the top for the sack of clarity.